sklearn/preprocessing/imputation.py

```diff
@@ -372,4 +375,13 @@ def transform(self, X):
 
             X[coordinates] = values
 
+            features_with_missing_values = np.where(np.any(mask, axis=0))[0]
+            imputed_mask = mask[:, features_with_missing_values]
+
+            if self.axis == 0:
+                self.imputed_features = valid_statistics_indexes[features_with_missing_values]
```

Comment:
I am not sure this line is logically correct; you do not currently test the case where a column is all masked.